### PR TITLE
Validate `AppleLocale` against supported locales

### LINKED ISSUE #28: Generating invalid locale
See https://github.com/zeit/hyper/issues/3091

It looks like if people have their region set to one country and the display language to another, os-locale generates a invalid xx_xx pair, not sure if this should be validated in the lib or the app using it.

### PR DESCRIPTION
**Pull request by pi0**
Fixes #28, zeit/hyper#3091, and https://github.com/zeit/hyper/pull/2913#issuecomment-447627297.

MacOS `defaults read -g AppleLocale` returns invalid POSIX values when the locale is different than language, causing values like `en_DE` or `en_IR`. Fortunately, we also have `/usr/bin/locale -a` which returns a list of supported locales.

This PR fixes issue, by validating locale against supporting locales and falling back to the `defaultLocale` in case of invalid one.

--- 

Attachment. Perl warning:

```
perl: warning: Setting locale failed.
perl: warning: Please check that your locale settings:
        LC_ALL = (unset),
        LANG = "en_DE.UTF-8"
    are supported and installed on your system.
perl: warning: Falling back to the standard locale ("C").
```

### CODE REVIEW

**Comment by pi0** [comment]
@sindresorhus Tests seem to fail on master too. Opened another PR for it.

File: index.js [hunk 1]
```diff
@@ -24,12 +24,30 @@ function getLocale(string) {
 	return (string && string.replace(/[.:].*/, ''));
 }
 
+function getLocales() {
+	return execa.stdout('locale', ['-a']);
```
Comment (sindresorhus) [inline index.js]: This can be cached globally as it won't change during the process lifetime.

File: index.js [hunk 2]
```diff
@@ -24,12 +24,30 @@ function getLocale(string) {
...
+}
+
+function getLocalesSync() {
+	return execa.sync('locale', ['-a']).stdout;
+}
+
+function getSupportedLocale(locale, locales = '') {
+	return locales.indexOf(locale) === -1 ? defaultLocale : locale;
```
Comment (sindresorhus) [inline index.js]: `indexOf` => `includes`

**Comment by sindresorhus** [comment]
Can you fix the merge conflict?

File: index.js (hunk 2 above)
Comment (pi0) [reply index.js]: I made this change. According to the [package.json](https://github.com/sindresorhus/os-locale/blob/master/package.json#L13) we require engine `>=6` however from [node.green](https://node.green/#ES2015-built-ins-well-known-symbols-Symbol-match--String-prototype-includes), this function is not available in `6.4.0`. However, users should always prefer the latest semver minior/patch of each release, i just mentioned this :)

**Comment by pi0** [comment]
@sindresorhus Thanks for time for review. I made changes and also added a comment.

**Pull request merged**